### PR TITLE
CFN publishing automation status fix

### PR DESCRIPTION
**Pull request by maastha**
## Description

Currently the automation document used to publish CFN resources shows the execution as "SUCCEEDED" even if build step fails.
We are adding the script to repo so that we can track changes between versions temporarily.

The changes have been tested by publishing cfn-resources (ThirdPartyIntegration) to AWS public registry.

Link to any related issue(s): 

## Type of change:

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [X] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

### CODE REVIEW

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (maastha) [inline cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:245]: Main change is in this step. Have committed current version in the branch for reference.

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:255]: [q] Does this work also when selecting multiple regions?

File: cfn-resources/cfn-publishing-automation/README.md
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/README.md:20]: Could you explain the process needed to update the script on aws? 

It would be great to explain that once the PR has been approved and merged, we should also update the script on AWS as follows: ....

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:44]: are these still needed?

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:32]: this description is not correct. This is the name of the project in CoderBuild

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:211]:
```suggestion
                - export ATLAS_ORG_ID={{OrgID}}
                - export ATLAS_PUBLIC_KEY={{PubKey}}
                - export ATLAS_PRIVATE_KEY={{PvtKey}}
                - export ATLAS_OUTPUT=json
```

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:250]: This `[0]` in particular let me think that may work only for one region

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:255]: yes, system manger creates one execution per region. This status is per execution.

File: cfn-resources/cfn-publishing-automation/README.md
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/README.md:20]: Yeah, for now I purposely didn't add much instructions since this is basically for internal use only. But I can add that.

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:44]: my understanding is yes, because even though resources don't use these anymore but the test scripts do when running tests during publishing

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:211]: some tests still use MCLI_* variables, those will need to be updated first, I'll list those out

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:255]: You can think of this document running separately for each region that we specify when creating the executions.

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:211]: Resource tests using MCLI_* variables:

Cloud-provider-access
Custom-dns-config-cluster-aws
Federated-settings-org-role
Network-container
Private-endpoint-regional-mode

File: cfn-resources/cfn-publishing-automation/README.md
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/README.md]: updated

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml]: updated

File: cfn-resources/cfn-publishing-automation/README.md
Comment (colm-quinn) [inline cfn-resources/cfn-publishing-automation/README.md:1]: 💯 Thank you for the readme - very useful!

File: cfn-resources/cfn-publishing-automation/README.md
Comment (colm-quinn) [inline cfn-resources/cfn-publishing-automation/README.md:25]:
```suggestion
To update the script in AWS you will need to create a new version of the document. We aim to keep a version in source control at all times which matches the currently deployed version. Please follow
```

File: cfn-resources/cfn-publishing-automation/README.md
Comment (colm-quinn) [inline cfn-resources/cfn-publishing-automation/README.md:36]: [q] I really like the version approach here, but it's quite manual and will I can imagine it will easily fall out of sync.

Have you checked is there any way to source the script from a git file or link? This would be common for IaC approaches - curious why AWS doesn't support it.

Comment (colm-quinn) [reply cfn-resources/cfn-publishing-automation/README.md:36]: Looks like they might have a python or ansible support: https://docs.aws.amazon.com/systems-manager/latest/userguide/integration-remote-scripts.html

This isn't blocking the PR, more a future improvement, but if we're going to use Evergreen in the future, probably overkill.

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for the readme improvements here.

For future, I think this would have been better as two PRs - it was tricky to review the actual change outside of adding the new script.

PR1: Add the script base version
PR2: Add the changes

File: cfn-resources/cfn-publishing-automation/README.md
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/README.md:36]: I think there are ways to use automation document from an S3 bucket but not from a git repo/link. Another way would be to use AWS SDK for Systems Manager to create and run a automation from our machine, probably then we'll be able to maintain the code in our repo (needs more investigation)

Comment (maastha) [reply cfn-resources/cfn-publishing-automation/README.md:36]: We could try this https://docs.aws.amazon.com/systems-manager/latest/userguide/run-remote-documents.html

Comment (colm-quinn) [reply cfn-resources/cfn-publishing-automation/README.md:36]: sgtm, something to consider when we move to Evergreen

**Review by colm-quinn — DISMISSED** [review]
LGTM

File: cfn-resources/cfn-publishing-automation/README.md
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/README.md:28]: Could you add a link to the AWS doc on how to add this new document?

File: cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml
Comment (andreaangiolillo) [inline cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml:216]: this is already done in `./cfn-install-mongo-cli.sh`
```suggestion
```

File: cfn-resources/cfn-publishing-automation/README.md
Comment (maastha) [reply cfn-resources/cfn-publishing-automation/README.md]: added link

**Review by colm-quinn — APPROVED** [review]

**Pull request merged**
